cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,237 @@
+	// 	logrus.WithFields(logrus.Fields{"calculation": newCalc.Name, "teff": teff, "logG": "4.0"}).Info("Creating calculation for test purpose")
+	// 	_, err := fakeClient.Calculations().Create(ctx, newCalc, metav1.CreateOptions{})
+	// 	if err != nil {
+	// 		t.Errorf("Coudn't create the calculation: %v", newCalc)
+	// 	}
+	// }
+
+	s := server{
+		logger: logrus.WithField("test-name", "get calculations test"),
+		ctx:    ctx,
+		client: fakeClient,
+	}
+
+	var jsonStr = []byte(`{"teff": "10000.0", "logG": "4"}`)
+	req, err := http.NewRequest("POST", "/calculations/create", bytes.NewBuffer(jsonStr))
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(s.createCalculation)
+	handler.ServeHTTP(rr, req)
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+	expected := `{"teff": "10000.0", "logG": "4"}`
+	// got := map[string]interface{}{
+	// 	"teff":
+	// }
+	if rr.Body.String() != expected {
+		t.Errorf("handler returned unexpected body: got %v want %v",
+			rr.Body.String(), expected)
+	}
+}
+
+func TestDeleteCalculation(t *testing.T) {
+	// type picked struct {
+	// 	DBKey string
+	// 	teff  string
+	// 	logG  string
+	// }
+
+	// testCases := []struct {
+	// 	id   string
+	// 	data []picked
+	// }{
+	// 	{
+	// 		id: "two calculations get displayed",
+	// 		data: []picked{
+	// 			{
+	// 				teff:  "10001",
+	// 				logG:  "4",
+	// 				DBKey: "vz.star:teff_10001",
+	// 			},
+	// 			{
+	// 				teff:  "10002",
+	// 				logG:  "4",
+	// 				DBKey: "vz.star:teff_10002",
+	// 			},
+	// 		},
+	// 	},
+	// }
+
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	teff := 10000
+	testCalculationCount := teff + 2
+	var dryCalcList []*calculationsv1.Calculation
+	for teff != testCalculationCount {
+		teff++
+		newCalc := util.NewCalculation(float64(teff), 4.0)
+		newCalc.DBKey = fmt.Sprintf("vz.star:teff_%d", teff)
+
+		dryCalcList = append(dryCalcList, newCalc)
+		logrus.WithFields(logrus.Fields{"calculation": newCalc.Name, "teff": teff, "logG": "4.0"}).Info("Creating calculation for test purpose")
+		_, err := fakeClient.Calculations().Create(ctx, newCalc, metav1.CreateOptions{})
+		if err != nil {
+			t.Errorf("Coudn't create the calculation: %v", newCalc)
+		}
+	}
+
+	s := server{
+		logger: logrus.WithField("test-name", "get calculations test"),
+		ctx:    ctx,
+		client: fakeClient,
+	}
+
+	req, err := http.NewRequest("DELETE", "/calculations/delete/{id}", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(s.deleteCalculation)
+	handler.ServeHTTP(rr, req)
+	if status := rr.Code; status != http.StatusOK {
```

Comment:
ok so here this will pass since we don't pass the statusCode in the headers. Can you fix that in this PR too? In every error that we return in all functions, please set `w.WriteHeader(statusCode)` as well.